src/bashi/filter_software_dependency.py

```diff
@@ -40,4 +43,85 @@ def software_dependency_filter(
+                if UBUNTU in row and row[UBUNTU].version >= pkv.parse("20.04"):
+                    return False
+
+    for compiler in (HOST_COMPILER, DEVICE_COMPILER):
+        if compiler in row and row[compiler].name == GCC:
+            if row[compiler].version <= pkv.parse("6"):
+                if UBUNTU in row and row[UBUNTU].version >= pkv.parse("20.04"):
+                    return False
+                        
+    if UBUNTU in row and row[UBUNTU].version >= pkv.parse("20.04"):
+        if HOST_COMPILER in row and DEVICE_COMPILER in row:
+         if GCC in (row[HOST_COMPILER].name, row[DEVICE_COMPILER].name):
+             if row[HOST_COMPILER].version <= pkv.parse("6") and row[DEVICE_COMPILER].version <= pkv.parse("6"): 
+                    return False
+                
+    for compiler in (HOST_COMPILER, DEVICE_COMPILER):
+        if compiler in row and row[compiler].name == GCC:
+            if row[compiler].version <= pkv.parse("6"):
+                if UBUNTU in row and row[UBUNTU].version >= pkv.parse("20.04"):
+                    return False
+
+    if UBUNTU in row and row[UBUNTU].version >= pkv.parse("20.04"):
+        if DEVICE_COMPILER in row and row[DEVICE_COMPILER].name == GCC:
+            if HOST_COMPILER in row and row[HOST_COMPILER].name == GCC:
+                for gcc_version in range(1,7):
+                    if row[HOST_COMPILER].version == gcc_version or row[DEVICE_COMPILER].version == gcc_version:
+                        return False
+
+    if HOST_COMPILER in row and DEVICE_COMPILER in row:
+        if GCC in (row[HOST_COMPILER].name, row[DEVICE_COMPILER].name) and row[DEVICE_COMPILER].version <= pkv.parse("6"):
+            if UBUNTU in row and row[UBUNTU].version >= pkv.parse("20.04"):
+                return False
+
+
+                   
+    for compiler in (HOST_COMPILER, DEVICE_COMPILER):
+        if compiler in row and row[compiler].name == GCC and row[compiler].version <= pkv.parse("6") and UBUNTU in row and row[UBUNTU].version >= pkv.parse("20.04"):
+                    return False
+
+    
+
+
+                    reason(
+                        output,
+                        "GCC as host and device compiler older than 6 version is not supported on Ubuntu 20.04 or newer",
+                    )
+                    
+                    
+
+"""
+    if UBUNTU in row and row[UBUNTU].version >= pkv.parse("20.04"):
+        if DEVICE_COMPILER in row and row[DEVICE_COMPILER].name == GCC:
+            if HOST_COMPILER in row and row[HOST_COMPILER].name == GCC:
+                for gcc_version in range(1, 7):
+                    if (
+                        row[HOST_COMPILER].version == gcc_version
+                        or row[DEVICE_COMPILER].version == gcc_version
+                    ):
+                        reason(
+                            output,
+                            "GCC as host and device compiler older than 6 version is not supported on Ubuntu 20.04 or newer",
```

Comment:
```suggestion
                            f"GCC {row[DEVICE_COMPILER].version} is not available in Ubuntu {__ubuntu_version_to_string(row[UBUNTU].version)}",",
```

`__ubuntu_version_to_string` is defined as follow:
```python
def __ubuntu_version_to_string(version: pkv.Version) -> str:
    return f"{version.major}.{version.minor:02}"
```